### PR TITLE
Expand support for http helpers for REST tests

### PR DESCRIPTION
**Pull request by ryanjdew**
add convenience functions (and unit test) for the HTTP POST/PUT calls and securely store credentials in Security database for use with REST calls.

### CODE REVIEW

File: ml-unit-test-client/build.gradle [hunk 1]
```diff
@@ -64,4 +64,49 @@ if (project.hasProperty("myBintrayUser")) {
...
+  String xquery = """
+xquery version "1.0-ml";
+
+import module namespace sec = "http://marklogic.com/xdmp/security"
+      at "/MarkLogic/security.xqy";
+
+declare option xdmp:mapping "false";
+
+try {
+  sec:remove-credential("ml-unit-test-credentials")
```
Comment (rjrudin) [inline ml-unit-test-client/build.gradle]: Does the Manage PUT endpoint not work if the credential already exists?

File: ml-unit-test-client/src/test/ml-modules/root/test/suites/More Unit Tests/http-helpers.xqy [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+
+test:assert-http-get-status("v1/search?format=xml", $test:DEFAULT_HTTP_OPTIONS,200),
```
Comment (dmcassel) [inline ml-unit-test-client/src/test/ml-modules/root/test/suites/More Unit Tests/http-helpers.xqy]: I don't see a definition for $test:DEFAULT_HTTP_OPTIONS.

File: ml-unit-test-client/build.gradle (hunk 1 above)
Comment (ryanjdew) [reply ml-unit-test-client/build.gradle]: Unfortunately, the REST endpoint only supports AWS authentication according to the documentation.

File: ml-unit-test-client/src/test/ml-modules/root/test/suites/More Unit Tests/http-helpers.xqy (hunk 2 above)
Comment (ryanjdew) [reply ml-unit-test-client/src/test/ml-modules/root/test/suites/More Unit Tests/http-helpers.xqy]: Thanks for catching that. I was in the client directory when I committed and pushed, so it missed my modules changes. I pushed the modules changes now.

**Pull request merged**